zebra/interface.c

```diff
@@ -1156,6 +1156,94 @@ static const char *zebra_ziftype_2str(zebra_iftype_t zif_type)
 	}
 }
 
+/* Interface's brief information print out to vty interface. */
+static void ifs_dump_brief_vty(struct vty *vty, struct vrf *vrf)
+{
+	struct connected *connected;
+	struct listnode *node;
+	struct route_node *rn;
+	struct zebra_if *zebra_if;
+	struct prefix *p;
+	struct interface *ifp;
+	bool print_header = true;
+
+	FOR_ALL_INTERFACES (vrf, ifp) {
+		bool is_global_v6 = false;
+		char global_pfx[PREFIX_STRLEN] = {0};
+		char ll_pfx[PREFIX_STRLEN] = {0};
+		bool break_loop = false;
+		char buf[PREFIX_STRLEN] = {0};
+
+		if (print_header) {
+			vty_out(vty, "%-8s%-16s%-16s%-20s%-51s\n", "status",
+				"ifname", "vrf", "ipv4", "ipv6");
+			vty_out(vty, "%-8s%-16s%-16s%-20s%-51s\n", "------",
+				"------", "---", "----", "----");
+			print_header = false; /* We have at least 1 iface */
+		}
+		zebra_if = ifp->info;
+
+
+		if (if_is_up(ifp))
+			vty_out(vty, "%-8s", "up");
+		else
+			vty_out(vty, "%-8s", "down");
+
+		vty_out(vty, "%-16s%-16s", ifp->name, vrf->name);
+
+		for (rn = route_top(zebra_if->ipv4_subnets); rn;
+		     rn = route_next(rn)) {
+			if (!rn->info)
+				continue;
+
+			for (ALL_LIST_ELEMENTS_RO((struct list *)rn->info, node,
+						  connected)) {
+				if (!CHECK_FLAG(connected->flags,
+						ZEBRA_IFA_SECONDARY)) {
+					p = connected->address;
+					prefix2str(p, buf, sizeof(buf));
+					vty_out(vty, "%-20s", buf);
+					/* break the outer loop as well */
+					break_loop = true;
+					break;
+				}
+			}
+			if (break_loop)
```

Comment:
- Now printing 1 address per subnet for ipv4. However, for ipv6 I don't see anything like `zebra_if->ipv4_subnets` in the code. Look at function "if_dump_vty". In if_dump_vty, we loop twice to print ipv4 but only once to print ipv6. Why? 

- Instead of adding the count, adding a "+" sign for additional addresses per subnet.

- Not printing IPv6 link-locals now. However, I am considering link-local to decide the count for the subnet hence displaying a "+" is dependent on link-local along with the global ipv6 address. Do let me know if this needs to be changed. We will always see a + sign for ipv6 because of this reason.

Sample output - 

```localhost.localdomain# show interface brief 
Interface       Status  VRF             Addresses
---------       ------  ---             ---------
enp0s3          up      default         10.0.2.15/24
                                        + 2a07:2ec0:2185:20:a824:7e41:87d9:eaac/64
lo              up      default         
virbr0          up      default         192.168.122.1/24
virbr0-nic      down    default         13.1.1.2/32
                                        13.1.1.3/32
                                        + 15.1.1.2/24
```